### PR TITLE
Implemented story MAT-523 

### PR DESCRIPTION
**Pull request by cday-semanticbits**
SDtory MAT-523: https://jira.cms.gov/browse/MAT-523
- Removed run char and dictionary word requirements from passwords.
- Password changed to 15 min chars 128 max chars.
- Updated password requirements text.
- Updated test cases for above changes.
- Added time to TRANSACTION_AUDIT_LOG entry to stop the server error because its a non null field in the table.
- Updated README.md with README_SB.md.
- Updated README.md with fixes I discovered when setting up the MAT.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,192 +1,165 @@
-[![Total alerts](https://img.shields.io/lgtm/alerts/g/MeasureAuthoringTool/MeasureAuthoringTool.svg?logo=lgtm&logoWidth=18)](https://lgtm.com/projects/g/MeasureAuthoringTool/MeasureAuthoringTool/alerts/) [![Language grade: Java](https://img.shields.io/lgtm/grade/java/g/MeasureAuthoringTool/MeasureAuthoringTool.svg?logo=lgtm&logoWidth=18)](https://lgtm.com/projects/g/MeasureAuthoringTool/MeasureAuthoringTool/context:java)
-
-# Measure Authoring Tool
+# Measure Authoring Tool (MAT)
 
 ## Installation
-The Measure Authoring Tool (MAT) was designed using many open source products including the Google Web Toolkit (GWT) framework, Java JDK, MySQL and Eclipse. GWT allows a developer to write client side code in Java and GWT converts it to JavaScript. The MAT uses MySQL as its backend database server and the IDE is Eclipse for Java EE Developers.
+The Measure Authoring Tool (MAT) was designed using many open source products including, the Google Web Toolkit (GWT)
```
Comment (serhii-ilin) [inline README.md]: I think that Andrew wanted to merge README and README_SB files.. 
The one with SB suffix has priority over regular README

File: README.md [hunk 2]
```diff
@@ -1,192 +1,165 @@
...
+- Go to System Preferences/ MY SQL after installing.
+- Click Initialize Database.
+- Enter a passowrd and click on Use Legacy Password Encryption. (Remember the username/pwd you will need these in future steps.)
+- Start the MYSQL Db.
+- In MeasureAuthoringTool/pom.xml I had to change the driver to match the community version I downloaded. I am always careful not to commit this change.
+  
 ```
+      <dependency>
+          <groupId>mysql</groupId>
+          <artifactId>mysql-connector-java</artifactId>
```
Comment (serhii-ilin) [inline README.md]: I think this trick is needed only if a developer uses MySQL ver  <8.
If you install recommended MySQL 5.7, then this is not needed.

File: README.md (hunk 1 above)
Comment (cday-semanticbits) [reply README.md]: This one is actually the merged one. README_SB.md was deleted.

File: src/test/java/mat/shared/PasswordVerifierTest.java [hunk 3]
```diff
@@ -1,40 +1,67 @@
 package mat.shared;
 
-import junit.framework.TestCase;
+import org.junit.Test;
```
Comment (serhii-ilin) [inline src/test/java/mat/shared/PasswordVerifierTest.java]: interesting...
You rely on Jnit5 assertions, but JUnit5 '@Test' annotation.
I would suggest making it consistent and use annotation from org.junit.jupiter.api
Not critical.

Comment (cday-semanticbits) [reply src/test/java/mat/shared/PasswordVerifierTest.java]: Good catch will change that.

File: README.md (hunk 1 above)
Comment (serhii-ilin) [reply README.md]: oh, great

File: src/test/java/mat/shared/PasswordVerifierTest.java (hunk 3 above)
Comment (cday-semanticbits) [reply src/test/java/mat/shared/PasswordVerifierTest.java]: Good catch. Will change it to use Assert.

File: README.md [hunk 4]
```diff
@@ -1,192 +1,165 @@
...
-            <artifactId>commons-collections</artifactId>
-            <groupId>commons-collections</groupId>
-        </exclusion>
-        <exclusion>
-            <groupId>org.eclipse.jetty</groupId>
-            <artifactId>apache-jsp</artifactId>
-        </exclusion>
-    </exclusions>
-</dependency>
+- Install MySQL (MAT currently has been tested with MySQL Community Version 8.0+) available from [MySQL] (https://dev.mysql.com/downloads/mysql/)
```
Comment (serhii-ilin) [inline README.md]: I think it was "MySQL (MAT currently has been tested with MySQL Community Version **5.7**)" 
I'm not aware of plans to upgrade MySQL to 8.0+
Do you have a confirmation from @AndrewBird81  or @nicole-hunter ?

File: README.md [hunk 5]
```diff
@@ -1,192 +1,165 @@
...
+Copy mysql jdbc to tomcat lib directory.
 ```
-cd /path/to/bonnie
-git secrets --install
-git secrets --register-aws
+cp ~/.m2/repository/mysql/mysql-connector-java/5.1.6/mysql-connector-java-5.1.6.jar /apache-tomcat-9.X.X/lib
 ```
+#### Build project with maven
+
+ `mvn clean install` you should see BUILD SUCCESS
```
Comment (serhii-ilin) [inline README.md]: Could you please add a requirement for the maven version? Required maven version is set to  3.6.3

File: README.md (hunk 4 above)
Comment (cday-semanticbits) [reply README.md]: I am not checking in the driver changes for the pom.xml actually included in the directions not to check that in. These are just instructions on how to run it locally. I gave up trying to get 5.7 working in osx. The latest version works if you change the driver but obviously I haven't done end to end testing on it.

File: README.md [hunk 6]
```diff
@@ -1,192 +1,165 @@
...
+  
 ```
+      <dependency>
+          <groupId>mysql</groupId>
+          <artifactId>mysql-connector-java</artifactId>
+          <version>8.0.19</version>
+      </dependency>
+```      
+- Create a new MySQL Connection to the database. I use jetbrains (datagrip)[https://www.jetbrains.com/datagrip/]
+- From the MAT Code base, find the `scripts/Dump*.sql` file and then execute the script in the database that was just created. <br> **(Note this script is from a dump and drops and create a schema called  `MAT_APP_BLANK`)**
```
Comment (serhii-ilin) [inline README.md]: The current dump name is different. It's Dump_MAT_APP_BLANK_V6.sql
Also, there is another recommended dump on a shared google drive https://drive.google.com/drive/u/0/folders/1x0WhhIM9WIwCzXtmm46iF6wjXeVui5ct

File: README.md (hunk 4 above)
Comment (serhii-ilin) [reply README.md]: Hm... interesting.. I was able to install MySQL 5.7.28 with brew on my Mac. I don't recall any issues...

File: README.md [hunk 7]
```diff
@@ -1,192 +1,188 @@
...
-3. Follow the instructions for setting up the pre-commit hooks (from https://github.com/awslabs/git-secrets):
+4.  Follow these [instructions](https://github.com/awslabs/git-secrets) for setting up the pre-commit hooks:
 
-```
-cd /path/to/bonnie
-git secrets --install
-git secrets --register-aws
-```
+    ```bash
+    cd /path/to/bonnie
```
Comment (adongare) [inline README.md]: It's not because of your change but was wrongly documented earlier. this should be path/to/MeasureAuthoringTool

Comment (cday-semanticbits) [reply README.md]: Changed it.

**Pull request merged**